### PR TITLE
WIP BXMSDOC-5309-master: Add new "Branches" section of endpoints for the Knowledge Store REST API

### PR DESCRIPTION
**Pull request by hmanwani-rh**
[RHPAM](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-5309-RHPAM/#knowledge-store-rest-api-branches-ref_kie-apis)
[RHDM](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-5309-RHDM/#knowledge-store-rest-api-branches-ref_kie-apis)

Added Branches section of endpoints for knowledge store REST API in Kie-API doc

### CODE REVIEW

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 1]
```diff
@@ -0,0 +1,220 @@
+// To reuse this module, ifeval the title to be more specific as needed.
+
+[id='knowledge-store-rest-api-branches-ref_{context}']
+= Branches
+
+The Knowledge Store REST API supports the following endpoints for managing branches in {CENTRAL}. The Knowledge Store REST API base URL is `\http://SERVER:PORT/{URL_COMPONENT_CENTRAL}/rest/`. All requests require HTTP Basic authentication or token-based authentication for the `rest-all` user role.
+
+[POST] /branches/{name}/maven/compile::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Referring to what has been stated below "_Compiles a specified branch in a specified project._",
the correct URL here should be `/spaces/{spaceName}/projects/{projectName}/branches/{name}/maven/compile`.
Another option is to make it clear in the introduction of this section (7.2.4.) that `/spaces/{spaceName}/projects/{projectName}` is part of the URL for all branch operations listed below. 

In any case, we have two additional parameters:
- spaceName: space name (required - string)
  /spaces/{`spaceName`}/projects/{projectName}/branches/{name}/maven/compile
- projectName: project name (required - string)
  /spaces/{spaceName}/projects/{`projectName`}/branches/{name}/maven/compile

Although `/spaces/{spaceName}/projects/{projectName}` has been discussed in the previous sections, in my opinion, it is important to make it clear to the user that the full URL is composed of: Base URL + Space part + Project part + Branch part.

For example:
`http://SERVER:PORT/{URL_COMPONENT_CENTRAL}/rest/spaces/{spaceName}/projects/{projectName}/branches/{name}/maven/compile`

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+{
+    "jobId":       "1576175811233-4",
+    "status":      "APPROVED",
+    "spaceName":   "Space123",
+    "projectName": "ProjABC",
+    "branchName":  "b1",
+}
+----
+--
+[POST] /branches/{name}/maven/install::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Same thing here (refer to my first comment).

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+{
+    "jobId":       "1576175811233-4",
+    "status":      "APPROVED",
+    "spaceName":   "Space123",
+    "projectName": "ProjABC",
+    "branchName":  "b1",
+}
+----
+--
+[POST] /branches/{name}/maven/test::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Same thing here (refer to my first comment).

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+{
+    "jobId":       "1576175811233-4",
+    "status":      "APPROVED",
+    "spaceName":   "Space123",
+    "projectName": "ProjABC",
+    "branchName":  "b1",
+}
+----
+--
+[POST] /branches/{name}/maven/deploy::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Same thing here (refer to my first comment).

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+{
+    "jobId":       "1576175811233-4",
+    "status":      "APPROVED",
+    "spaceName":   "Space123",
+    "projectName": "ProjABC",
+    "branchName":  "b1",
+}
+----
+--
+[GET] /branches::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Same thing here (refer to my first comment).

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+[source,json]
+----
+[
+  {
+    "name":"master"
+  }
+]
+----
+--
+[POST] /branches::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Same thing here (refer to my first comment).

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    "status":         "APPROVED",
+    "spaceName":      "Space123",
+    "projectName":    "ProjABC",
+    "newBranchName":  "b1",
+    "baseBranchName": "master",
+    "userIdentifier": "bc"
+}
+----
+--
+[DELETE] /branches/{name}::
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Same thing here (refer to my first comment).

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|*body*
+|The `newBranchName` and `baseBranchName` of a project
+|String
```
Comment (caponetto) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc:233]:
```suggestion
|Request body
```
Other tables are referring to types like this as `Request body`.

**Review by caponetto — CHANGES_REQUESTED** [review]

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+  {
+    "name":"master"
+  }
+]
+----
+--
+[POST] /spaces/{spaceName}/projects/{projectName}/branches::
++
+--
+Adds branch in a specified project and space.
```
Comment (sterobin) [inline doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Suggest "Adds a specified branch in ..." for consistency with the others and clarity.

**Review by sterobin — COMMENTED** [review]
@hmanwani-rh , just one comment. Otherwise good. Excellent work on this Heena! You did an exceptional job following the pattern and voice of the doc. Doesn't look at all like a new writer took it over, which is the ideal. Thanks so much.

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc (hunk 9 above)
Comment (hmanwani-rh) [reply doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Thank you Stetson

**Comment by barboras7** [comment]
@hmanwani-rh,
looks good to me. One small suggestion - maybe it would be worth mentioning that if no branch is specified, the maven actions (test, compile, install & deploy) apply to the master branch 

For example:
`http://localhost:8080/business-central/rest/spaces/{spaceName}/projects/{projectName}/maven/install`

CC @caponetto, what do you think?

**Comment by caponetto** [comment]
+1 for @barboras7 suggestion

**Review by caponetto — APPROVED** [review]

File: doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc (hunk 1 above)
Comment (hmanwani-rh) [reply doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/knowledge-store-rest-api-branches-ref.adoc]: Thanks Guilherme for providing the inputs.

**Pull request merged**